test/setup-and-deploy.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+# This script includes common functions for setting up different Gateway API implementations.
+source "$(dirname $0)"/e2e-common.sh
+
+test_setup
```

Comment:
the `hack` scripts already call this - you shouldn't need to

ie. see lifecycle: https://github.com/knative/hack#using-the-e2e-testssh-helper-script